src/substrait/textplan/parser/SubstraitPlanRelationVisitor.cpp

```diff
@@ -238,6 +271,49 @@ std::string normalizeProtoEnum(std::string_view text, std::string_view prefix) {
   return result;
 }
 
+void addInputFieldsToSchema(
+    RelationType relationType,
+    std::shared_ptr<RelationData>& relationData) {
+  if (relationData->continuingPipeline != nullptr) {
+    auto continuingRelationData = ANY_CAST(
+        std::shared_ptr<RelationData>, relationData->continuingPipeline->blob);
+    if (!continuingRelationData->outputFieldReferences.empty()) {
+      for (auto field : continuingRelationData->outputFieldReferences) {
+        relationData->fieldReferences.push_back(field);
+      }
+    } else {
```

Comment:
Maybe add a comment that this is "with emit" and "without emit"?

Reply:
Done.